Share Target with single quote throws an error

The [issue](https://github.com/pwa-builder/PWABuilder/issues/2812) was raised on the [PWABuilder](https://pwabuilder.com/) repo. We use bubblewrap to package TWAs. It seems to throw the following error every time we try to create a package, and it's due to the french characters being used in the apps manifest, specifically in the description:

> "description": "La Chorale de Cocody Philadelphie est une chorale l'Eglise Adventiste du 7ième jour de Cocody Philadelphie. Elle chante pour la gloire de DIEU et pour gagner des âmes à Christ.",

cli ERROR Command failed: gradlew.bat assembleRelease --stacktrace

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\jaylynbarbee\OneDrive - Microsoft\Desktop\bbwrap-test\app\build.gradle' line: 80

* What went wrong:
Could not compile build file 'C:\Users\jaylynbarbee\OneDrive - Microsoft\Desktop\bbwrap-test\app\build.gradle'.
> startup failed:
build file 'C:\Users\jaylynbarbee\OneDrive - Microsoft\Desktop\bbwrap-test\app\build.gradle': 80: unexpected char: '\' @ line 80, column 430.
pour gagner des �mes � Christ.\\",\\"url

**To Reproduce**
Steps to reproduce the behavior:
1. bubblewrap init --manifest https://adventiste-ccp.web.app/manifest.webmanifest
2. bubblewrap build

**Expected behavior**
Package generated with no error.

--- a/packages/core/src/spec/lib/utilSpec.ts
+++ b/packages/core/src/spec/lib/utilSpec.ts
@@ -306,13 +306,14 @@ describe('util', () => {
   });
 
   describe('#escapeJsonString combined with #escapeGradleString returns the expected results',
-   () => {
-    it ('Escapes double and single quotes', () => {
-      // String.raw prevents escape characters from being applied, so we can simplify howe we write
-      // the expected value.
-      let expected = String.raw`\\"Hello Worl\\\'d\\"`;
-      expect(util.escapeJsonString(util.escapeGradleString(`"Hello Worl'd"`)))
-          .toEqual(expected);
-    });
-  });
+      () => {
+        it('Escapes double and single quotes', () => {
+          // String.raw prevents escape characters from being applied, so we can simplify howe we write
+          // the expected value.
+          const input = String.raw`"Hello Worl'd"`;
+          const expected = String.raw`\\"Hello Worl\\\'d\\"`;
+          expect(util.escapeJsonString(util.escapeGradleString(input)))
+              .toEqual(expected);
+        });
+      });
 });

--- a/packages/core/src/spec/lib/utilSpec.ts
+++ b/packages/core/src/spec/lib/utilSpec.ts
@@ -304,4 +304,14 @@ describe('util', () => {
           .toEqual('\\"\\$Hello\\\\Wo\\`eld\\"');
     });
   });
+
+  describe('#escapeJsonString combined with #escapeGradleString returns the expected results', () => {
+    it ('Escapes double and single quotes', () => {
+      // String.raw prevents escape characters from being applied, so we can simplify howe we write
+      // the expected value.
+      let expected = String.raw`\\"Hello Worl\\\'d\\"`;
+      expect(util.escapeJsonString(util.escapeGradleString(`"Hello Worl'd"`)))
+          .toEqual(expected);
+    });
+  });
 });

--- a/packages/core/src/spec/lib/utilSpec.ts
+++ b/packages/core/src/spec/lib/utilSpec.ts
@@ -308,7 +308,7 @@ describe('util', () => {
   describe('#escapeJsonString combined with #escapeGradleString returns the expected results',
       () => {
         it('Escapes double and single quotes', () => {
-          // String.raw prevents escape characters from being applied, so we can simplify howe we write
+          // String.raw prevents escape characters from being applied, so we can simplify how we write
           // the expected value.
           const input = String.raw`"Hello Worl'd"`;
           const expected = String.raw`\\"Hello Worl\\\'d\\"`;

--- a/packages/core/src/spec/lib/utilSpec.ts
+++ b/packages/core/src/spec/lib/utilSpec.ts
@@ -305,7 +305,8 @@ describe('util', () => {
     });
   });
 
-  describe('#escapeJsonString combined with #escapeGradleString returns the expected results', () => {
+  describe('#escapeJsonString combined with #escapeGradleString returns the expected results',
+   () => {
     it ('Escapes double and single quotes', () => {
       // String.raw prevents escape characters from being applied, so we can simplify howe we write
       // the expected value.

--- a/packages/core/template_project/app/build.gradle
+++ b/packages/core/template_project/app/build.gradle
@@ -77,7 +77,7 @@ android {
 
         <% if (shareTarget) { %>
             // The data for the app to support web share target.
-            resValue "string", "shareTarget", '<%= escapeJsonString(JSON.stringify(shareTarget)) %>'
+            resValue "string", "shareTarget", '<%= escapeJsonString(escapeGradleString(JSON.stringify(shareTarget))) %>'
         <% } %>
 
         // The hostname is used when building the intent-filter, so the TWA is able to

